File: coding/Codeforces/round_129/task_A.scala
/*
https://codeforces.com/contest/1681/problem/A
 */
//package round_129

import scala.collection.mutable
import scala.io.StdIn

object task_A extends App {
    def solving(x: Array[Int], y: Array[Int]) = {
        val max_x = x.max
        val max_y = y.max
        if (max_x > max_y) {
            1
        }
        else if (max_x < max_y) {
            -1
        }
        else {
            0
        }
    }

    var sb = new StringBuilder()
    val t = StdIn.readLine().toInt
    for (i <- 0 until(t)) {
        val n = StdIn.readLine().toInt
        val a = StdIn.readLine().split(" ").map(_.toInt)
//        val a = Array(n)
//        for (ii <- 0 until(n)) a(ii) = StdIn.readLine().split(" ").toInt
        val m = StdIn.readLine().toInt
        val b = StdIn.readLine().split(" ").map(_.toInt)
//        val b = Array(m)
//        for (ii <- 0 until(m)) b(ii) = StdIn.readLine().toInt
        val ans = solving(a, b)
        if (ans == 1) {
//            println("Alice")
//            println("Alice")

            sb.append("Alice").append("\n").append("Alice").append("\n")
        }
        else if (ans == -1) {
//            println("Bob")
//            println("Bob")

            sb.append("Bob").append("\n").append("Bob").append("\n")
        }
        else {
//            println("Alice")
//            println("Bob")

            sb.append("Alice").append("\n").append("Bob").append("\n")

        }
    }

    print(sb)
}

File: coding/Codeforces/VK_cup_22/task_A.cpp
// https://codeforces.com/contest/1769/problem/A
// Created by ruslan515 on 06.12.22.
//

#include <iostream>

int main() {
    int n;
    std::cin >> n;
    int a[n];
    for (int i = 0; i < n; ++i) {
        std::cin >> a[i];
    }

    int ans[n];
    ans[0] = a[0] - 1;
    std::cout << ans[0] << "\n";
    for(int i = 1; i < n; ++i) {
        int delta = a[i] - (i + 1);
        if (delta <= ans[i - 1]) {
            delta = ans[i - 1] + 1;
        }
        ans[i] = delta;
        std::cout << ans[i] << "\n";
    }


    return 0;
}